### PR TITLE
Integrating indexers from go-commons

### PR DESCRIPTION
**Pull request by vishnuchalla**
### Description
Integrating with indexers in go-commons

### Testing
```
time="2023-04-29 23:13:14" level=info msg="📁 Creating indexer: elastic" file="metrics.go:39"
time="2023-04-29 23:13:16" level=info msg="👽 Initializing prometheus client with URL: https://prometheus-k8s-openshift-monitoring.apps.vchalla-quay-test-2.perfscale.devcluster.openshift.com" file="prometheus.go:55"
time="2023-04-29 23:13:18" level=info msg="Scraping for the prometheus entry with params - Endpoint: https://prometheus-k8s-openshift-monitoring.apps.vchalla-quay-test-2.perfscale.devcluster.openshift.com, Profile: /home/vchalla/quay-metrics.yaml, Start: 2023-04-29T22:13:14-04:00, End: 2023-04-29T23:13:14-04:00" file="utils.go:67"
time="2023-04-29 23:13:18" level=info msg="Indexing metrics with UUID test-uuid" file="utils.go:72"
time="2023-04-29 23:13:18" level=info msg="🔍 Scraping prometheus metrics for benchmark from 2023-04-29T22:13:14-04:00 to 2023-04-29T23:13:14-04:00" file="prometheus.go:115"
time="2023-04-29 23:13:18" level=info msg="Indexing metric QuayRequestDuration" file="prometheus.go:145"
time="2023-04-29 23:13:25" level=info msg="Indexing finished in 6.249s: created=121" file="prometheus.go:151"
time="2023-04-29 23:13:25" level=info msg="Indexing metric QuaySecScanIndexDuration" file="prometheus.go:145"
time="2023-04-29 23:13:30" level=info msg="Indexing finished in 4.776s: created=121" file="prometheus.go:151"
time="2023-04-29 23:13:30" level=info msg="Indexing metric QuayModelCacheTotalRate" file="prometheus.go:145"
time="2023-04-29 23:13:30" level=info msg="Indexing finished in 0s:" file="prometheus.go:151"
time="2023-04-29 23:13:30" level=info msg="Indexing metric QuayModelCacheTotalSum" file="prometheus.go:145"
time="2023-04-29 23:13:30" level=info msg="Indexing finished in 0s:" file="prometheus.go:151"
time="2023-04-29 23:13:30" level=info msg="Indexing metric QuayDbConnectCallsRate" file="prometheus.go:145"
time="2023-04-29 23:13:34" level=info msg="Indexing finished in 3.934s: created=121" file="prometheus.go:151"
time="2023-04-29 23:13:35" level=info msg="Indexing metric QuayDbConnectCallsSum" file="prometheus.go:145"
time="2023-04-29 23:13:39" level=info msg="Indexing finished in 4.117s: created=121" file="prometheus.go:151"
time="2023-04-29 23:13:39" level=info msg="Indexing metric QuayDbCloseCallsRate" file="prometheus.go:145"
time="2023-04-29 23:13:45" level=info msg="Indexing finished in 6.07s: created=121" file="prometheus.go:151"
time="2023-04-29 23:13:46" level=info msg="Indexing metric QuayDbCloseCallsSum" file="prometheus.go:145"
time="2023-04-29 23:13:51" level=info msg="Indexing finished in 4.499s: created=121" file="prometheus.go:151"
time="2023-04-29 23:13:51" level=info msg="Indexing metric QuayDbPooledConnectionsAvailableRate" file="prometheus.go:145"
time="2023-04-29 23:13:57" level=info msg="Indexing finished in 5.235s: created=121" file="prometheus.go:151"
time="2023-04-29 23:13:57" level=info msg="Indexing metric QuayDbPooledConnectionsAvailableSum" file="prometheus.go:145"
time="2023-04-29 23:14:03" level=info msg="Indexing finished in 6.006s: created=121" file="prometheus.go:151"
time="2023-04-29 23:14:03" level=info msg="Indexing metric QuayDbPooledConnectionsInUseRate" file="prometheus.go:145"
time="2023-04-29 23:14:08" level=info msg="Indexing finished in 4.8s: created=121" file="prometheus.go:151"
time="2023-04-29 23:14:10" level=info msg="Indexing metric QuayDbPooledConnectionsInUseSum" file="prometheus.go:145"
time="2023-04-29 23:14:15" level=info msg="Indexing finished in 5.45s: created=121" file="prometheus.go:151"
time="2023-04-29 23:14:15" level=info msg="👋 Exiting kube-burner test-uuid" file="kube-burner.go:198"
```

### CODE REVIEW

File: pkg/measurements/factory.go [hunk 1]
```diff
@@ -20,8 +20,8 @@ import (
 
 	"github.com/cloud-bulldozer/kube-burner/log"
 	"github.com/cloud-bulldozer/kube-burner/pkg/config"
-	"github.com/cloud-bulldozer/kube-burner/pkg/indexers"
 	"github.com/cloud-bulldozer/kube-burner/pkg/measurements/types"
+	"github.com/vishnuchalla/go-commons/indexers"
```
Comment (jtaleric) [inline pkg/measurements/factory.go:24]: Did we want to use the `cloud-bulldozer` one?

Comment (vishnuchalla) [reply pkg/measurements/factory.go:24]: Yes we will ultimately do that once, https://github.com/cloud-bulldozer/go-commons/pull/1 is merged. If we do it now itself the `go build` is going to fail as the `cloud-bulldozer` repo is empty as of now.

File: go.mod [hunk 2]
```diff
@@ -1,16 +1,16 @@
 module github.com/cloud-bulldozer/kube-burner
 
-go 1.19
```
Comment (rsevilla87) [inline go.mod:3]: 1.18?

File: cmd/kube-burner/kube-burner.go [hunk 3]
```diff
@@ -32,8 +32,8 @@ import (
 	"github.com/cloud-bulldozer/kube-burner/pkg/util/metrics"
 	"github.com/cloud-bulldozer/kube-burner/pkg/version"
 
-	"github.com/cloud-bulldozer/kube-burner/pkg/indexers"
 	"github.com/cloud-bulldozer/kube-burner/pkg/prometheus"
+	"github.com/vishnuchalla/go-commons/indexers"
```
Comment (rsevilla87) [inline cmd/kube-burner/kube-burner.go:36]: We can replace this import now

Comment (vishnuchalla) [reply cmd/kube-burner/kube-burner.go]: Yes changed and updated accordingly. Thanks!

File: pkg/burner/metadata.go [hunk 4]
```diff
@@ -48,5 +49,12 @@ func indexjobSummaryInfo(indexer *indexers.Indexer, uuid string, elapsedTime flo
 			Version:     fmt.Sprintf("%v@%v", version.Version, version.GitCommit),
 		},
 	}
-	(*indexer).Index(metadataInfo, indexers.IndexingOpts{MetricName: jobSummaryMetric, JobName: jobConfig.Name})
+	log.Infof("Indexing metric %s", jobSummaryMetric)
+	log.Debugf("Indexing [%d] documents in %s", len(metadataInfo), indexName)
+	resp, err := (*indexer).Index(metadataInfo, indexers.IndexingOpts{MetricName: jobSummaryMetric, JobName: jobConfig.Name})
+	if err != nil {
+		log.Fatal(err.Error())
```
Comment (rsevilla87) [inline pkg/burner/metadata.go:56]: We shouldn't fail the execution on indexing errors

File: pkg/alerting/alert_manager.go [hunk 5]
```diff
@@ -203,7 +202,14 @@ func parseMatrix(value model.Value, description string, severity severityLevel)
...
 	log.Info("Indexing alerts")
-	(*a.indexer).Index(alertSet, indexers.IndexingOpts{MetricName: alertMetricName})
+	log.Infof("Indexing metric %s", alertMetricName)
+	log.Debugf("Indexing [%d] documents in %s", len(alertSet), indexName)
+	resp, err := (*a.indexer).Index(alertSet, indexers.IndexingOpts{MetricName: alertMetricName})
+	if err != nil {
+		log.Fatal(err.Error())
+	} else {
+		log.Debug(resp)
+	}
```
Comment (rsevilla87) [inline pkg/alerting/alert_manager.go:214]:
```suggestion
	  if err != nil {
			log.Error(err)
		} else {
			log.Info(resp)
		}
```

File: pkg/alerting/alert_manager.go [hunk 6]
```diff
@@ -203,7 +202,14 @@ func parseMatrix(value model.Value, description string, severity severityLevel)
 	return result, alertSet, nil
 }
 
-func (a *AlertManager) index(alertSet []interface{}) {
+func (a *AlertManager) index(alertSet []interface{}, indexName string) {
 	log.Info("Indexing alerts")
-	(*a.indexer).Index(alertSet, indexers.IndexingOpts{MetricName: alertMetricName})
+	log.Infof("Indexing metric %s", alertMetricName)
+	log.Debugf("Indexing [%d] documents in %s", len(alertSet), indexName)
```
Comment (rsevilla87) [inline pkg/alerting/alert_manager.go:208]: Let's stop printing the indexName

File: pkg/prometheus/prometheus.go [hunk 7]
```diff
@@ -140,8 +140,16 @@ func (p *Prometheus) ScrapeJobsMetrics(indexer *indexers.Indexer) error {
...
+		indexerConfig := p.ConfigSpec.GlobalConfig.IndexerConfig
+		if indexerConfig.Enabled {
+			log.Infof("Indexing metric %s", md.MetricName)
+			log.Debugf("Indexing [%d] documents in %s", len(datapoints), indexerConfig.Index)
+			resp, err := (*indexer).Index(datapoints, indexers.IndexingOpts{MetricName: md.MetricName})
+			if err != nil {
+				log.Fatal(err.Error())
+			} else {
+				log.Debug(resp)
+			}
```
Comment (rsevilla87) [inline pkg/prometheus/prometheus.go:152]:
```suggestion
	if err != nil {
		log.Error(err)
	} else {
		log.Info(resp)
	}
```

File: pkg/util/metrics/tarball.go [hunk 8]
```diff
@@ -95,7 +94,11 @@ func ImportTarball(tarball string, indexer *indexers.Indexer) error {
 			return fmt.Errorf("Tarball read error: %v", err)
 		}
 		log.Infof("Importing metrics from %s", hdr.Name)
-		(*indexer).Index(metrics, indexers.IndexingOpts{})
+		log.Infof("Writing metric to: %s", metricsDir)
+		_, err = (*indexer).Index(metrics, indexers.IndexingOpts{})
+		if err != nil {
+			log.Fatal(err.Error())
```
Comment (rsevilla87) [inline pkg/util/metrics/tarball.go:100]:
```suggestion
			return err
```

File: vendor/github.com/opensearch-project/opensearch-go/OpenSearch.svg
Comment (rsevilla87) [inline vendor/github.com/opensearch-project/opensearch-go/OpenSearch.svg:1]: lol, we have to remove vendor asap

File: pkg/prometheus/prometheus.go (hunk 7 above)
Comment (rsevilla87) [reply pkg/prometheus/prometheus.go:152]: We shouldn't error on indexing error

File: pkg/alerting/alert_manager.go (hunk 5 above)
Comment (vishnuchalla) [reply pkg/alerting/alert_manager.go:214]: Changed as suggested.

File: pkg/measurements/vmi_latency.go [hunk 9]
```diff
@@ -379,8 +379,23 @@ func (p *vmiLatency) stop() (int, error) {
...
+	metricMap := map[string][]interface{}{
+		podLatencyMeasurement:          p.normLatencies,
+		podLatencyQuantilesMeasurement: p.latencyQuantiles,
+	}
+	for metricName, data := range metricMap {
+		indexingOpts.MetricName = metricName
+		log.Debugf("Indexing [%d] documents", len(data))
+		resp, err := (*factory.indexer).Index(data, indexingOpts)
+		if err != nil {
+			log.Error(err)
```
Comment (rsevilla87) [inline pkg/measurements/vmi_latency.go:394]:
```suggestion
			log.Error(err.Error())
```

**Review by rsevilla87 — APPROVED** [review]
lgtm!

**Pull request merged**